cfpb/cfpb-chart-builder
# New choropleth map

This adds a `geo-map` chart type that displays a choropleth map. Its API is the same as the other charts.

The chart's `metadata` field is used to control the charts geographic granularity. It can show data at the national, state or county level. The shapefiles are not bundled with this library because they can be quite large. Instead, they're loaded asynchronously on demand.

## Additions

- `geo-map` chart type

## Removals

-

## Changes

-

## Testing

- Pull down this branch.
- `nvm use 8`
- `./setup.sh` to build everything
- `npm test` to verify unit and browser tests pass as normal (you'll need to add [sauce credentials](https://github.com/cfpb/cfpb-chart-builder#testing) if you haven't already
- `gulp watch` and look at the new map at the top of the page. After five seconds it will toggle back and forth between state and county views to demonstrate its `update()` method.

To verify the charts still load correctly in cfgov-refresh:

- Pull down this branch.
- `nvm use 8`
- `npm link`
- `cd ~/wherever/your/projects/are/cfgov-refresh/`
- `npm link cfpb-chart-builder`
- `gulp build`
- `./runserver`
- Verify that [this page](http://127.0.0.1:8000/data-research/consumer-credit-trends/credit-cards/borrower-risk-profiles/) and [this page](https://www.consumerfinance.gov/data-research/consumer-credit-trends/credit-cards/borrower-risk-profiles/) are identical and that the charts aren't borked.

[Preview this PR without the whitespace changes](?w=0)

## Screenshots

<img width="675" alt="screen shot 2017-08-14 at 1 42 27 am" src="https://user-images.githubusercontent.com/1060248/29259618-e2858eea-8091-11e7-903c-1b32841c8bd7.png">

## Notes

-

## Todos

- Colors! We don't want the green and the border colors need to be refined. I need to look into how to assign color values via CSS. The tile map does this so it shouldn't be too bad.
- Puerto Rico needs to be added to the shapefiles.
- The dividing line between Alaska and Hawaii needs to be improved.

## Checklist
